superset/forms.py

```diff
@@ -66,8 +108,14 @@ def csv_enabled_dbs():
             FileRequired(), FileAllowed(['csv'], _('CSV Files Only!'))])
     con = QuerySelectField(
         _('Database'),
-        query_factory=csv_enabled_dbs,
+        query_factory=csv_allowed_dbs,
         get_pk=lambda a: a.id, get_label=lambda a: a.database_name)
+    schema = StringField(
+        _('Schema'),
+        description=_('Specify a schema (if database flavour supports this).'),
+        validators=[Optional()],
+        widget=BS3TextFieldWidget(),
+        filters=[lambda x: x or None])
```

Comment:
Did not notice that. I just copy-pasted to move these lines of code for a better order showing on the UI. Will take a look.

Reply:
Took a quick look. I think the original author wrote this because that if no `schema` is provided on the frontend, the backend will receive an empty string which is `''`, but it should be `None` otherwise the database will throw some kind of error like `schema does not exist`. So yes, I think it is just casting falsy values, specifically `''` to `None`.